### ISSUE TITLE
Unclear how to install this plugin how I expect to use javascript libraries

### ISSUE THREAD

**Issue by wschenk** [KyleAMathews/react-markdown-textarea#11]
In your instructions you say you use this by saying “npm install react-markdown-textarea”.  I’m building a project on middleman and using bower for javascript dependancies.  I’m not sure how to get to the javascript in this case.  (This is my ignorance of the backend javascript world, I’m used to rails/sprockets.)  Do you have any pointers on how I get this to work?

**Comment by KyleAMathews** [comment]
I've written this module with the expectation that you'll use https://www.npmjs.com/ for managing your dependencies as it's *much* easier for both me as a library developer and you the end user to work with.

Browserify or Webpack are two bundler tools that take javascript from NPM and turn it into browser-ready JS.

I'd suggest learning one of the above two tools and integrating it into your workflow as React is *much* easier to work with when using commonjs modules and most open-sourced React components (like this one) assume you'll be using one of the above.

**Comment by wschenk** [comment]
Ah, ok, is there anyway to pull it apart so I can use it directly?  Like I said, I'm already using middleman/sprockets for many other reasons other than the javascript part so changing build systems isn't realistic.

**Comment by KyleAMathews** [comment]
You can use Browserify at least with middleman/sprockets I believe. If that
fails, you can generate a standalone version of the component.

Are you using React?

**Comment by wschenk** [comment]
Yes, I already have a react site built, I'm using middleman/sprockets to generate the site, and all of my React components are in separate cjsx files.  What I'd like is to have just the TextareaMarkdown class be in it's own file, with no dependancies, and attaching the class to the global namespace.  It seems like that's pretty much the opposite of what the goals of browserify or webpack are, so I just rewrote it.

Also, I'm a little confused on the code that you have in there.  It has a .js extension, but it looks like it's being compiled by babel and so it's actually .es6?

**Comment by KyleAMathews** [comment]
Which code? The actual src code for the module is at https://github.com/KyleAMathews/react-markdown-textarea/blob/master/src/index.cjsx

And sorry for not being more clear but when I said "you can generate a standalone version" I meant you can generate a global version of this module. E.g. here are instructions for Browserify http://www.forbeslindesay.co.uk/post/46324645400/standalone-browserify-builds

**Comment by wschenk** [comment]
Thank you!  That helps explain things to me.  I need to figure out a way to learn about how javascript is put together in this crazy new world!

**Comment by KyleAMathews** [comment]
:)

There's definitely a learning curve but the "crazy new world" is pretty fun and wonderfully productive. There's a ton of goodies on NPM that are very easily accessible once you get the hang of it.
